k4GeneratorsConfig/src/xsection.cxx

```diff
@@ -84,25 +84,26 @@ bool k4GeneratorsConfig::xsection::processFile(){
   }
   unsigned int lastEvent = m_reader->getEntries(podio::Category::Event) - 1;
   auto event = podio::Frame(m_reader->readEntry(podio::Category::Event,lastEvent));
-  
+
   // decode sqrts
-  m_sqrts = event.getParameter<double>("SQRTS");
+  const auto& sqrts = event.getParameter<double>("SQRTS");
+  m_sqrts = sqrts.value_or(0.);
 
   // decode the cross sections
   bool readOK = true;
-  std::vector<double> xsections = event.getParameter<std::vector<double>>("CrossSections");
-  if ( xsections.size() > 0 ){
-    m_xsection =  xsections[0];
+  const auto& xsections = event.getParameter<std::vector<double>>("CrossSections");
+  if ( xsections.has_value() ){
+    m_xsection =  xsections.value()[0];
   }
   else {
     m_xsection =  0.;
     readOK = false;
   }
-  
```

Comment:
Some of these constructs could now be refactored into the following, sincy you can actually check whether the parameter has been set
```cpp
// If you are only intersted in the first value in any case, you can simply get it via
const auto maybeXsection = event.getParameter<double>("CrossSections");
m_xsection = maybeXsection.value_or(0.0);
readOK = maybeXsection.has_value();
```